src/main/java/org/wildfly/security/auth/server/event/RealmIdentitySuccessfulAuthorizationEvent.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2015 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wildfly.security.auth.server.event;
+
+import org.wildfly.security.authz.AuthorizationIdentity;
+
+/**
+ * An event indicating that one identity attempted to authorize as another identity.
+ *
+ * @author <a href="mailto:david.lloyd@redhat.com">David M. Lloyd</a>
+ */
+public final class RealmIdentitySuccessfulAuthorizationEvent extends RealmIdentityAuthorizationEvent {
+    private final AuthorizationIdentity newAuthorizationIdentity;
```

Comment:
On reflection, I don't like this.  This potentially allows an authorization identity from another realm to get passed to the target realm.  I think I'll change this to be more superficial, like old/new Principal or something.